src/index.ts

```diff
@@ -0,0 +1,108 @@
+import * as core from "@actions/core";
+import { File as DiffFile } from "parse-diff";
+import minimatch from "minimatch";
+import { callModel, createReviewPrompt, parseAIResponse } from "./ai";
+import { ReviewComment } from "./types";
+import {
+  getFileContent,
+  getPRContext,
+  getRepoContext,
+  submitReviewComments,
+} from "./git";
+
+// TODO: add conditional review when added as reviewer
+// TODO: customize the bot name and avatar
+// TODO: set up cicd to deploy the action
+
+async function prepareReviewContext(
+  prContext: Awaited<ReturnType<typeof getPRContext>>
```

Comment:
Consider using a more specific type for `prContext` instead of `Awaited<ReturnType<typeof getPRContext>>`. This would make the function signature more readable and maintainable.